### ISSUE TITLE
Nesting rule is broken?

### ISSUE THREAD

**Issue by mzdr** [sasstools/sass-lint#771]
Hey there,

I'm using `grunt-sass-lint` (v0.2.0) with `sass-lint` (v1.8.2) and I'm having troubles with the nesting rule. There is this specific pattern where the rule isn't working anymore.

I'm using `nesting-depth` with `max-depth: 1`, the whole configuration file is attached. ([sass-lint.yml.txt](https://github.com/sasstools/sass-lint/files/340897/sass-lint.yml.txt))

So here is what I do:

``` css
.a {
    content: "root";

    .x1 {
        content: "first level - without this, it works";
    }

    .b {
        content: "first level";

        .x2 {
            content: "second level - without this, it works";
        }

        .c {
            content: "second level";

            .x3 {
                content: "third level - without this, it works";
            }

            .d {
                content: "third level";

                .e {
                    content: "fourth level";
                }
            }
        }
    }
}
```

This piece of code is **not** being reported by sass-lint with `max-depth` set to `1`, **but** as soon as I remove one of the `.xN` classes, the linting is reporting that nesting is wrong. But I am expecting it to report `.x2` and `.c` straight from the beginning.

The output when I remove `.x1` for example:

> src/main/sass/styles.scss
>  56:9  warning  Nesting depth 2 greater than max of 1  nesting-depth
>  60:9  warning  Nesting depth 2 greater than max of 1  nesting-depth

When `.x1` is still there, sass-lint will be quiet. I tried an IDE plugin as well as the CLI.

**Comment by adeacetis** [comment]
Hello @mzdr,
We've been investigating this in one of our project for a couple of days. We just realized that this rule is working as expected.

You and us are looking for something different, the `SelectorDepth` rule which is/was available in scss-lint. Unless I missed it, it remains to be added  to the project.

Here's a bit of doc. about the `SelectorDepth`:
https://github.com/brigade/scss-lint/tree/master/lib/scss_lint/linter#selectordepth

Is that what you are looking for? 

Cheers,
Axel
